azurelinuxagent/ga/monitor.py

```diff
@@ -253,13 +262,14 @@ def collect_and_send_events(self):
 
     def daemon(self):
         min_delta = min(MonitorHandler.TELEMETRY_HEARTBEAT_PERIOD,
-                        MonitorHandler.CGROUP_TELEMETRY_PERIOD,
+                        MonitorHandler.CGROUP_TELEMETRY_POLLING_PERIOD,
+                        MonitorHandler.CGROUP_TELEMETRY_REPORTING_PERIOD,
                         MonitorHandler.EVENT_COLLECTION_PERIOD,
                         MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD,
                         MonitorHandler.IMDS_HEARTBEAT_PERIOD).seconds
         while self.should_run:
             self.send_telemetry_heartbeat()
-            self.send_cgroup_telemetry()
+            self.send_extension_metrics_telemetry()
```

Comment:
minor: name needs to be a little more generic, since this is for any cgroups (extensions, agent, services)

Reply:
The `extension` in `send_extension_metrics_telemetry` was targetted more on the extension as a package (which could depend on other things on the box as well like services), rather than only focusing on extensions runtime. It could in the future send not only cgroups but extension status, python package information, etc.

Reply:
if we add more data, it is likely the reporting period will be different. I would make this specific to resource metrics